receiver/kafkareceiver/kafka_receiver.go

```diff
@@ -126,8 +127,8 @@ func createKafkaClient(config Config) (sarama.ConsumerGroup, error) {
 	return sarama.NewConsumerGroup(config.Brokers, config.GroupID, saramaConfig)
 }
 
-func (c *kafkaTracesConsumer) Start(_ context.Context, _ component.Host) error {
-	ctx, cancel := context.WithCancel(context.Background())
+func (c *kafkaTracesConsumer) Start(ctx context.Context, _ component.Host) error {
```

Comment:
I can remove this change if necessary since it's unrelated to the bug fix, but the receiver should be creating the child context from the passed in context, not `context.Background`. Let me know if this should be removed.

Reply:
From the [component.Start](https://github.com/open-telemetry/opentelemetry-collector/blob/main/component/component.go#L28:L40) method:

> 	// If the component needs to perform a long-running starting operation then it is recommended
	// that Start() returns quickly and the long-running operation is performed in background.
	// In that case make sure that the long-running operation does not use the context passed
	// to Start() function since that context will be cancelled soon and can abort the long-running
	// operation. Create a new context from the context.Background() for long-running operations.

I worry this works by exiting quickly and doesn't run for the complete time. Can I ask you to confirm that this approach works and the advice provided by the type is out of date?

Reply:
I'll test when I get a chance, but definitely not worth going against the spec for this change. Thanks for bringing this up @MovieStoreGuy, I hadn't seen that in the spec!

Reply:
I was able to test, it looks like using the passed in context works today when running the collector from the CLI, since the collector is starting components synchronously in the same goroutine. The context being passed in isn't cancelled within the lifespan of this component, as the child context created by `WithCancel` is cancelled during `Shutdown` before the passed in context is closed.

Still worth removing to conform to the spec though, I could imagine relying on `context.Background` being useful in different system architectures down the road.